### PR TITLE
Rename two MGs

### PR DESCRIPTION
**Pull request by GiudGiud**
not the most influential PR of course.

But I believe by 'sorting' our objects better we display better what we can do for a given operation. It helps see the global picture of what's been done from simply looking at the list of objects with the same prefixes, and when we need a new object it s easier to know where to poach code from.

I think this beats having truly unique names that differentiate objects better, but get missed when looking at the list of objects, either on a syntax page or using the vscode auto-complete

### CODE REVIEW

**Comment by GiudGiud** [comment]
ugh someone broke registerMooseDeprecated

File: framework/src/meshgenerators/BreakMeshByBlockGenerator.C [hunk 1]
```diff
@@ -23,10 +23,10 @@ BreakMeshByBlockGenerator::validParams()
 {
   InputParameters params = BreakMeshByBlockGeneratorBase::validParams();
   params.addRequiredParam<MeshGeneratorName>("input", "The mesh we want to modify");
-  params.addClassDescription("Break boundaries based on the subdomains to which their sides are "
-                             "attached. Naming convention for the new boundaries will be the old "
-                             "boundary name plus \"_to_\" plus the subdomain name. At the moment"
-                             "this only works on REPLICATED mesh");
+  params.addClassDescription(
+      "Break the mesh at interfaces between blocks. New nodes will be generated so elements on "
+      "each side of the break are no longer connected. At the moment, this only works on "
```
Comment (lindsayad) [inline framework/src/meshgenerators/BreakMeshByBlockGenerator.C:28]:
```suggestion
      "each side of the break are no longer connected. At the moment, this only works on a "
```

Comment (GiudGiud) [reply framework/src/meshgenerators/BreakMeshByBlockGenerator.C]: done

**Review by roystgnr — APPROVED** [review]
ExplodeMesh->BreakMeshByElement is a big improvement, and ElementSubdomainID->SubdomainPerElement seems fine.

I'd say give people a couple days to catch up on their vacation notifications, and potentially disagree, before merging.

**Comment by lindsayad** [comment]
works for me

**Review by dschwen — APPROVED** [review]

**Pull request merged**